src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -1820,6 +1907,253 @@ public string DeleteCounter(DocumentsOperationContext context, string documentId
+                        {
+                            // we need to change the counter group key and delete the old one
+                            table.DeleteByKey(counterGroupKey);
+
+                            names.GetPropertyByIndex(1, ref prop);
+                            using var newScope = context.Allocator.Allocate(CounterKeysSlice.Size + 1 + 1, out ByteString newCounterKey);
+                            documentKeyPrefix.CopyTo(newCounterKey.Ptr);
+                            Memory.Copy(newCounterKey.Ptr + documentKeyPrefix.Size, prop.Name.Buffer, prop.Name.Size);
+                            Slice.From(context.Allocator, newCounterKey.Ptr, documentKeyPrefix.Size + prop.Name.Size, out counterGroupKey);
+                        }
+
+                        counterValues.Modifications ??= new DynamicJsonValue(counterValues);
+                        counterValues.Modifications.Remove(item.Name);
+
+                        names.Modifications ??= new DynamicJsonValue(names);
+                        names.Modifications.Remove(item.Name);
+
+                        data.Modifications = new DynamicJsonValue(data);
+                        using (var old = data)
+                        {
+                            data = context.ReadObject(data, item.DocumentId, BlittableJsonDocumentBuilder.UsageMode.ToDisk);
+                        }
+
+                        var newEtag = _documentsStorage.GenerateNextEtag();
+                        var newChangeVector = _documentsStorage.GetNewChangeVector(context, newEtag);
+
+                        using (Slice.From(context.Allocator, newChangeVector, out var cv))
+                        using (DocumentIdWorker.GetStringPreserveCase(context, collectionName.Name, out Slice collectionSlice))
+                        using (table.Allocate(out TableValueBuilder tvb))
+                        {
+                            tvb.Add(counterGroupKey);
+                            tvb.Add(Bits.SwapBytes(newEtag));
+                            tvb.Add(cv);
+                            tvb.Add(data.BasePointer, data.Size);
+                            tvb.Add(collectionSlice);
+                            tvb.Add(context.GetTransactionMarker());
+
+                            table.Set(tvb);
+                        }
+                    }
+
+                    try
+                    {
+                        var document = _documentsStorage.Get(context, item.DocumentId,
+                            throwOnConflict: true);
+
+                        var newData = document.Data.Clone(context);
+                        _documentDatabase.DocumentsStorage.Put(context, item.DocumentId, expectedChangeVector: null, newData, flags: document.Flags, nonPersistentFlags: document.NonPersistentFlags);
+                    }
+                    catch (Exception e)
+                    {
+                        throw new Exception($"Failed to get or update document '{item.DocumentId}'," +
```

Comment:
pass the entire `e` here as an inner exception and not just the message.. and make this exception as `InvalidOperationException`.

Reply:
done